api/rest/coordinator.go

```diff
@@ -0,0 +1,256 @@
+package rest
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+const numberOfWorkers = 10
+const contextDoneErrMsg = "bundle creation context finished before bundle creation finished"
+
+// BundleStatus tracks the status of local bundle creation requests
+type BundleStatus struct {
+	id   string
+	node node
+	done bool
+	err  error
+}
+
+// Coordinator is an interface to coordinate the creation of diagnostics bundles
+// across a cluster of nodes
+type Coordinator interface {
```

Comment:
I think this could be package private or we should move Coordinator to separate package
```suggestion
type coordinator interface {
```

Reply:
Interestingly, making this private causes our linter to fail. Since we only test `parallelCoordinator` as itself and not as a `coordinator` I think Go isn't bothering to consider that it implements the `coordinator` interface so it assumes it isn't used.